src_c/math.c

```diff
@@ -800,6 +804,80 @@ vector_copy(pgVector *self, PyObject *_null)
     return (PyObject *)ret;
 }
 
+static PyObject *
+vector_clamp_magnitude(pgVector *self, PyObject *args, PyObject *kwargs)
+{
+    Py_ssize_t i;
+    pgVector *ret;
+
+    ret = (pgVector *)pgVector_NEW(self->dim);
```

Comment:
Will this also need to be updated to use the new subtype changes in #3088 ?

Reply:
I plan on updating this in the subtype changes PR itself, unless the subtype PR gets merged first.